### ISSUE TITLE
Setup database to save backups in object storage

### ISSUE THREAD

**Issue by marcellmueller** [bcgov/CONN-CCBC-portal#436]
Currently we are backing up once a day and saving 30 days of backups but it's all on openshift which is a risk that can be mitigated if we can backup on AWS as well
- [ ] Update crunchy-postgres repo to support object storage in pgBackRest
- [x] Provision new s3 bucket
- [x] Adjust backup limit in helm chart - done in another ticket as PVC storage was increased
- [ ] Setup backups to AWS in helm chart

Old PR from before we realized PGO 5.3 was required that will be helpful to reference:
https://github.com/bcgov/CONN-CCBC-portal/tree/feat/db-aws-backup

**Comment by eazeredo** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/CCBC---Delivery-Board-61c0c1204c8bcb001491c5f3/issues/bcgov/CONN-CCBC-portal/436?planning-poker) with Zenhub @AntBush @marcellmueller @matthieu-foucault @VB-at-Bis @wenzowski

**Comment by VB-at-Bis** [comment]
Submitted PR https://github.com/bcgov/CONN-CCBC-portal/pull/561 for task #1.
Applied updated script and created s3 buckets in every environment.

**Comment by matthieu-foucault** [comment]
As far as I can tell, this is blocked on https://github.com/BCDevOps/developer-experience/issues/2536, see my comment: https://github.com/BCDevOps/developer-experience/issues/2536#issuecomment-1232420460

**Comment by matthieu-foucault** [comment]
With my testing and reading through the docs, I have confirmed that the postgres operator can handle having multiple backup locations. Although this does nothing to reduce the risk of a backup PVC running out of space, this would make releasing this feature post launch less daunting than anticipated. I would recommend to release the change in two steps:
1. add AWS backups
2. remove PVC backups once we've confirmed successful backups on AWS

**Comment by MarsRomer** [comment]
Blocked:  Mars to reach out to see if we can unblock or help with #2536

**Comment by rafasdc** [comment]
Reference docs: https://access.crunchydata.com/documentation/postgres-operator/latest/tutorials/backups-disaster-recovery/backups

Using the AWS integrated identity provider and role does not seem to work using our key and role combination and seem to be designed to Postgres clusters deployed on AWS EKS. 

Using the regular S3 Credentials with the same ones as our app gives an access denied error, likely due to our app using key + AWS ARN combination which does not seem to be supported.

**Comment by rafasdc** [comment]
This issue is blocked as regular IAM users are not supported, as per discussion with platform services: 

> Sorry. Regular IAM users are not supported in workload accounts. This IAM user that you are trying to use is a special user that is created in a special account and has a trust relationship with the role in the workload account

Other relevant info on this: 

The functionality was requested on this Issue https://github.com/CrunchyData/postgres-operator/issues/2509 but was implemented against web identity and assumes it is deployed inside AWS EKS. 

The "auto" option of pgbackrest is setup to do a loopback to 169.254.169.254 as it is designed to be ran inside an EC2 instance. 

Relevant pgbackrest code can be found here: https://github.com/pgbackrest/pgbackrest/blob/main/src/storage/s3/storage.c
